tests/Functional/BaseFunctionalTest.php

```diff
@@ -92,40 +95,33 @@ protected function getCommand(
             'token' => (string) getenv('MANAGE_API_TOKEN'),
         ]);
 
-        $jobFactory = new JobFactory(
+        $newJobFactory = new NewJobFactory(
             $storageClientFactory,
-            new JobFactory\JobRuntimeResolver($storageClientFactory),
-            $this->objectEncryptor,
-            new DataPlaneObjectEncryptorFactory(
-                (string) parse_url((string) getenv('STORAGE_API_URL'), PHP_URL_HOST),
-                (string) getenv('AWS_REGION'),
-            ),
-            new DataPlaneConfigRepository(
-                $manageApiClient,
-                new DataPlaneConfigValidator(Validation::createValidator())
+            new JobRuntimeResolver($storageClientFactory),
+            new DataPlaneObjectEncryptorProvider(
+                $this->objectEncryptor,
+                new DataPlaneConfigRepository(
+                    $manageApiClient,
+                    new DataPlaneConfigValidator(Validation::createValidator()),
+                    (string) getenv('ENCRYPTOR_STACK_ID'),
+                    (string) getenv('AWS_REGION'),
+                ),
+                false
             ),
-            false
         );
 
-        $job = $jobFactory->createNewJob($jobData);
+        $job = $newJobFactory->createNewJob($jobData);
         $queueClient = $this->getMockBuilder(QueueClient::class)
-            ->setMethods(['getJob', 'postJobResult', 'getJobFactory', 'updateJob', 'patchJob'])
+            ->setMethods(['getJob', 'postJobResult', 'updateJob', 'patchJob'])
             ->disableOriginalConstructor()
             ->getMock();
         $queueClient->expects(self::once())->method('getJob')->willReturn($job);
-        $queueClient->expects(self::any())->method('getJobFactory')->willReturn($jobFactory);
         $queueClient->expects(self::any())->method('updateJob')->willReturn([]);
         $queueClient->expects(self::any())->method('patchJob')->willReturn(
             new JobFactory\Job(
-                $this->objectEncryptor,
+                new JobObjectEncryptor($this->objectEncryptor),
                 $storageClientFactory,
-                [
-                    'runId' => '1234',
-                    'status' => 'processing',
-                    'projectId' => '123',
-                    'componentId' => 'dummy',
-                    'configId' => '123',
-                ]
+                array_merge($job->jsonSerialize(), ['status' => 'processing'])
```

Comment:
Tady ten test zacal failovat, protoze neco dal zkusilo pristoupit na `id`, ktery tu v tech datech neni. Ale kdyz jsem koukal, co to dela, prislo mi vlastne v poradku, z eby to melo vratit data toho stejnyho jobu, jen s jinym stavem.

Reply:
Mělo, to byla nějaká demence